groups/bsl/bslstl/bslstl_bitset.h

```diff
@@ -209,15 +227,167 @@ namespace bsl {
 template <std::size_t N>
 class bitset;
 
+                          // ====================
+                          // class Bitset_ImpUtil
+                          // ====================
+struct Bitset_ImpUtil {
+  enum {
+        k_BYTES_PER_INT = sizeof(int),
+        k_BITS_PER_INT  = CHAR_BIT * sizeof(int),
+        k_INTS_IN_LONG = sizeof(long) / sizeof(int)
+    };
+
+    static void defaultInit(unsigned int* data, size_t size, unsigned long val = 0);
+        // Initialize the memory at the address specified by 'data' so that the
+        // the first 'M' bit positions correspond to the bit values in the
+        // specified 'val' where 'M' is the smaller of 'size * k_BITS_PER_INT'
+        // and 'CHAR_BIT * sizeof(unsigned long)'.  The remaining bits are
+        // initialized to zero '0'.
+};
+
+void Bitset_ImpUtil::defaultInit(unsigned int *data,
+                                 size_t size,
+                                 unsigned long val)
+{
```

Comment:
You cannot define a non-template method in a header file unless you declare it inline, because you will get multiple-definition errors if the header file is included in multiple compilation units.